docs/book/src/reference/testing/integration.md

```diff
@@ -77,4 +77,62 @@ func TestReconcile(t *testing.T) {
\ No newline at end of file
+Both functions are located in `pkg/controller/parent/parent_controller_suite_test.go` file. The file also contains a `TestMain` function that allows you to specify CRD directory paths for the testing environment.
+
+## Using envtest in your integration tests
+[`controller-runtime`]() offers `envtest` ([godoc](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest)), a package that helps write integration tests for your controllers by setting up and starting an api-server binary and an etcd binary.
+
+Using `envtest` in integration tests follows the general flow of:
+
+```
+import sigs.k8s.io/controller-runtime/pkg/envtest
+
+//specify testEnv configuration
+testEnv = &envtest.Environment{
+	CRDDirectoryPaths: []string{filepath.Join("..", "config", "crd", "bases")},
+}
+
+//start testEnv
+cfg, err = testEnv.Start()
+
+//write test logic
+
+//stop testEnv
+err = testEnv.Stop()
+```
+
+`kubebuilder` does the boilerplate setup and teardown of testEnv for you, in the ginkgo test suite that it generates under the `/controllers` directory.
+
+Logs from the test runs are prefixed `test-env`.
+
+### Configuring your test control plane
+You can use environment variables and/or flags to specify the `api-server` and `etcd` setup within your integration tests.
+
+#### Environment Variables
+
+| Variable name | Type | When to use |
+| --- | :--- | :--- |
+| `USE_EXISTING_CLUSTER` | boolean | Instead of setting up a local control plane, point to the control plane of an existing cluster. |
+| `TEST_ASSET_KUBE_APISERVER`, `TEST_ASSET_ETCD`, `TEST_ASSET_KUBECTL` | paths to, respectively, api-server, etcd and kubectl binaries | Point integration tests to use binaries other than the default ones. They can also be used to ensure specific tests run with expected versions of these binaries.|
+| `KUBEBUILDER_CONTROLPLANE_START_TIMEOUT` and `KUBEBUILDER_CONTROLPLANE_STOP_TIMEOUT` | durations in seconds | Specify timeouts different from the default for the test control plane to (respectively) start and stop; any test run that exceeds them will fail. |
+| `KUBEBUILDER_ATTACH_CONTROL_PLANE_OUTPUT` | boolean | Set to `true` to attach the control plane's stdout and stderr to os.Stdout and os.Stderr. This can be useful when debugging test failures, as output will include output from the control plane. |
+| `KUBEBUILDER_ASSETS` | path to the `kubebuilder` binary | Use a different `kubebuilder` than the one in the machine's `PATH`. |
+
+#### Flags
+Here's an example of modifying the flags with which to start the API server in your integration tests:
+
+```
+exampleApiServerFlags := []string{
+	"--etcd-servers=<ip:port>",
+	"--cert-dir=/path/to/cert",
+	"--insecure-port=8081",
+	"--insecure-bind-address=localhost",
+	"--secure-port=6443",
+	"--admission-control=MutatingAdmissionWebhook",
+}
```

Comment:
Prob just append copy and then append to DefaultAPIServerFlags (the variable from envtest) here.

Reply:
👍 Changed [here](https://github.com/kubernetes-sigs/kubebuilder/pull/1055/commits/92c2875db101b684babb57b9559c6784fab83270).